x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/huggingface/rerank/HuggingFaceRerankModel.java

```
@@ -0,0 +1,91 @@
+    }
+
+    public HuggingFaceRerankModel(
+        String inferenceEntityId,
+        TaskType taskType,
+        String service,
+        Map<String, Object> serviceSettings,
+        Map<String, Object> taskSettings,
+        @Nullable Map<String, Object> secrets,
+        ConfigurationParseContext context
+    ) {
+        this(
+            inferenceEntityId,
+            taskType,
+            service,
+            HuggingFaceRerankServiceSettings.fromMap(serviceSettings, context),
+            HuggingFaceRerankTaskSettings.fromMap(taskSettings),
+            DefaultSecretSettings.fromMap(secrets)
+        );
+    }
+
+    // Should only be used directly for testing
+    public HuggingFaceRerankModel(
+        String inferenceEntityId,
+        TaskType taskType,
+        String service,
+        HuggingFaceRerankServiceSettings serviceSettings,
+        HuggingFaceRerankTaskSettings taskSettings,
+        @Nullable DefaultSecretSettings secrets
+    ) {
+        super(
+            new ModelConfigurations(inferenceEntityId, taskType, service, serviceSettings, taskSettings),
+            new ModelSecrets(secrets),
+            serviceSettings,
+            secrets
+        );
+    }
+
+    private HuggingFaceRerankModel(HuggingFaceRerankModel model, HuggingFaceRerankTaskSettings taskSettings) {
+        super(model, taskSettings);
+    }
+
+    @Override
+    public HuggingFaceRerankServiceSettings getServiceSettings() {
+        return (HuggingFaceRerankServiceSettings) super.getServiceSettings();
+    }
+
+    @Override
+    public HuggingFaceRerankTaskSettings getTaskSettings() {
+        return (HuggingFaceRerankTaskSettings) super.getTaskSettings();
+    }
+
+    @Override
+    public DefaultSecretSettings getSecretSettings() {
+        return (DefaultSecretSettings) super.getSecretSettings();
+    }
+
+    @Override
+    public Integer getTokenLimit() {
+        return getServiceSettings().maxInputTokens();
```

Let's just throw and unsupported exception like we do in Jan's PR: https://github.com/elastic/elasticsearch/pull/127254/files#r2071978538

We're using the GenericRequestManager so this method shouldn't be called.